File: src/main/java/cz/solutions/cockroach/ParsedExport.java
package cz.solutions.cockroach;

import lombok.Value;

import java.util.List;

@Value
public class ParsedExport {
    List<RsuRecord> rsuRecords;
    List<EsppRecord> esppRecords;
    List<DividendRecord> dividendRecords;
    List<TaxRecord> taxRecords;
    List<TaxReversalRecord> taxReversalRecords;
    List<SaleRecord> saleRecords;
    List<JournalRecord> journalRecords;
}

File: src/main/kotlin/cz/solutions/cockroach/ParsedExport.kt
package cz.solutions.cockroach

data class ParsedExport(
    val rsuRecords: List<RsuRecord>,
    val esppRecords: List<EsppRecord>,
    val dividendRecords: List<DividendRecord>,
    val taxRecords: List<TaxRecord>,
    val taxReversalRecords: List<TaxReversalRecord>,
    val saleRecords: List<SaleRecord>,
    val journalRecords: List<JournalRecord>,
    val interestRecords: List<InterestRecord> = emptyList()
) {
    operator fun plus(other: ParsedExport): ParsedExport {
        return ParsedExport(
            rsuRecords = rsuRecords + other.rsuRecords,
            esppRecords = esppRecords + other.esppRecords,
            dividendRecords = dividendRecords + other.dividendRecords,
            taxRecords = taxRecords + other.taxRecords,
            taxReversalRecords = taxReversalRecords + other.taxReversalRecords,
            saleRecords = saleRecords + other.saleRecords,
            journalRecords = journalRecords + other.journalRecords,
            interestRecords = interestRecords + other.interestRecords
        )
    }

    companion object {
        fun empty(): ParsedExport = ParsedExport(
            emptyList(), emptyList(), emptyList(),
            emptyList(), emptyList(), emptyList(), emptyList(),
            emptyList()
        )
    }
}